### ISSUE TITLE
Calling getUploadTasks for AWSS3TransferUtility gets all tasks

### ISSUE THREAD

**Issue by nick3389** [aws-amplify/aws-sdk-ios#2228]
**Describe the bug**
By calling the getUploadTasks method of a transfer utility after the app gets active to re associate the uncompleted tasks to a completion handler, this method returns even the completed tasks. I have confirmed it many times. This happens every time I put the app to the background and then open it in the foreground.

**To Reproduce**
Upload something, push the app to background and open the app again by calling getUploadTasks in applicationDidBecomeActive. All the tasks (even the finished) are returned.

**Which AWS service(s) are affected?**
AWSS3TransferUtility

**Expected behavior**
I would expect to get only the uncompleted tasks.

**Screenshots**

**Environment(please complete the following information):**
 - SDK Version: 2.12.5
 - Dependency Manager: Cocoapods
 - Swift Version : 5.1.3

**Device Information (please complete the following information):**
 - Device: iPhone 6s, Simulator
 - iOS Version: iOS 13.3
 - Specific to simulators: -

**Additional context**

**Comment by kneekey23** [comment]
hey @nick3389 can you post a sample of your code possibly?

**Comment by nick3389** [comment]
@kneekey23 When coming to foreground I call this:

`let uploadTasks: [AWSS3TransferUtilityUploadTask] = transferUtility?.getUploadTasks().result as? [AWSS3TransferUtilityUploadTask] ?? []`

and then I called:
`uploadTasks.forEach({$0.setCompletionHandler(completionBlockUpload!)})`

which caused the handler to be called again even for the completed tasks.

I put a temporary fix by replacing this line with:
`uploadTasks.filter({$0.progress.fractionCompleted < 1}).forEach({$0.setCompletionHandler(completionBlockUpload!)})`

**Comment by kneekey23** [comment]
hi @nick3389 thanks for the code snippet. What is the reason for setting the completion handler again after you have already set it previously? Can you try calling resume on the tasks when the app becomes active instead of resetting the completion handler?

**Comment by nick3389** [comment]
@kneekey23 Actually, that's what I was doing in the beginning, but at some point I wasn't notified that some files were uploaded and I look back in the documentation of AWSS3TransferUtility and I read this:

_Manage a Transfer when a Suspended App Returns to the Foreground

When an app that has initiated a transfer becomes suspended and then returns to the foreground, the transfer may still be in progress or may have completed. In both cases, use the following code to re-establish the progress and completion handler blocks of the app._

and the proposed code is:

`let uploadTasks = transferUtility.getUploadTasks().result
  for task in uploadTasks! {
       task.setCompletionHandler(completionHandler!)
       task.setProgressBlock(progressBlock!)
}`

That's why I changed my implementation and added the code I wrote in my previous comment. Is this correct or not?
